mlir/lib/Mitigation/Transforms/MitigationMethods/Zne.cpp

```diff
@@ -0,0 +1,310 @@
+    OpBuilder::InsertionGuard guard(rewriter);
+    ModuleOp moduleOp = op->getParentOfType<ModuleOp>();
+    Type qregType = quantum::QuregType::get(rewriter.getContext());
+
+    std::string fnAllocName = op.getCallee().str() + ".quantumAlloc";
+    if (moduleOp.lookupSymbol<func::FuncOp>(fnAllocName)) {
+        return SymbolRefAttr::get(ctx, fnAllocName);
+    }
+    Type i64Type = rewriter.getI64Type();
+    FunctionType fnAllocType = FunctionType::get(ctx, /*inputs=*/
+                                                 i64Type,
+                                                 /*outputs=*/qregType);
+    func::FuncOp fnAlloc = rewriter.create<func::FuncOp>(loc, fnAllocName, fnAllocType);
+    fnAlloc.setPrivate();
+    Block *allocBloc = fnAlloc.addEntryBlock();
+    rewriter.setInsertionPointToStart(allocBloc);
+    Value nQubits = allocBloc->getArgument(0);
+    IntegerAttr intAttr;
+    auto qreg = rewriter.create<quantum::AllocOp>(loc, qregType, nQubits, intAttr);
+    rewriter.create<func::ReturnOp>(loc, qreg.getResult());
+    return SymbolRefAttr::get(ctx, fnAllocName);
+}
+FlatSymbolRefAttr ZneLowering::getOrInsertFnWithoutMeasurements(Location loc,
+                                                                PatternRewriter &rewriter,
+                                                                mitigation::ZneOp op)
+{
+    MLIRContext *ctx = rewriter.getContext();
+    OpBuilder::InsertionGuard guard(rewriter);
+    ModuleOp moduleOp = op->getParentOfType<ModuleOp>();
+    Type qregType = quantum::QuregType::get(rewriter.getContext());
+    func::FuncOp fnOp = SymbolTable::lookupNearestSymbolFrom<func::FuncOp>(op, op.getCalleeAttr());
+    TypeRange originalTypes = op.getArgs().getTypes();
+
+    std::string fnWithoutMeasurementsName = op.getCallee().str() + ".withoutMeasurements";
+    if (moduleOp.lookupSymbol<func::FuncOp>(fnWithoutMeasurementsName)) {
+        return SymbolRefAttr::get(ctx, fnWithoutMeasurementsName);
+    }
+
+    SmallVector<Type> typesWithoutMeasurements = {originalTypes.begin(), originalTypes.end()};
+    typesWithoutMeasurements.push_back(qregType);
+
+    FunctionType fnWithoutMeasurementsType = FunctionType::get(ctx, /*inputs=*/
+                                                               typesWithoutMeasurements,
+                                                               /*outputs=*/qregType);
+    func::FuncOp fnWithoutMeasurementsOp =
+        rewriter.create<func::FuncOp>(loc, fnWithoutMeasurementsName, fnWithoutMeasurementsType);
+    fnWithoutMeasurementsOp.setPrivate();
+    rewriter.cloneRegionBefore(fnOp.getBody(), fnWithoutMeasurementsOp.getBody(),
+                               fnWithoutMeasurementsOp.end());
+    Block *fnWithoutMeasurementsBlock = &fnWithoutMeasurementsOp.front();
+    fnWithoutMeasurementsBlock->addArgument(qregType, loc);
+    quantum::AllocOp allocOp = *fnWithoutMeasurementsOp.getOps<quantum::AllocOp>().begin();
```

Comment:
I suppose :thinking: 
```suggestion
    quantum::AllocOp allocOp = fnWithoutMeasurementsOp->getOps<quantum::AllocOp>().begin();
```

Reply:
No it doe not work because I think the star is here to get the value of the begin() and front does not work